Sources/X509/OCSP/OCSPPolicy.swift

```diff
@@ -230,24 +268,95 @@ public struct OCSPVerifierPolicy<Requester: OCSPRequester>: VerifierPolicy, Send
+            }
+            
+            switch response.verifyTime(validationTime: self.validationTime) {
+            case .meetsPolicy:
+                break
+            case .failsToMeetPolicy(let reason):
+                return .failsToMeetPolicy(reason: reason)
+            }
+            
             switch response.certStatus {
-            case .good:
-                switch response.verifyTime(now: self.now ?? Date()) {
-                case .meetsPolicy:
-                    break
-                case .failsToMeetPolicy(let reason):
-                    return .failsToMeetPolicy(reason: reason)
-                }
-                
-                // TODO: verify signature: rdar://104687979
-                return .meetsPolicy
             case .revoked(let info):
                 return .failsToMeetPolicy(reason: "revoked through OCSP, reason: \(info.revocationReason?.description ?? "nil")")
             case .unknown:
                 return .failsToMeetPolicy(reason: "OCSP response returned as status unknown")
+            case .good:
+                return .meetsPolicy
             }
         }
     }
+    
+    private func validateResponseSignature(
+        _ basicResponse: BasicOCSPResponse,
+        issuer: Certificate
+    ) async -> PolicyEvaluationResult {
+        let responderID = basicResponse.responseData.responderID
+        let potentialCertificates = [issuer] + [basicResponse.certs?.first].compactMap { $0 }
+        let leafCertificate = potentialCertificates.first { certificate in
+            switch responderID {
+            case .byName(let subject):
+                return certificate.subject == subject
+            case .byKey(let subjectKeyIdentifier):
+                return (try? certificate.extensions.subjectKeyIdentifier?.keyIdentifier == subjectKeyIdentifier.bytes) ?? false
+            }
+        }
+        guard let leafCertificate else {
+            return .failsToMeetPolicy(reason: "could not find OCSP responder certificate for id \(responderID)")
+        }
+        
+        var verifier = Verifier(
+            rootCertificates: CertificateStore([issuer]),
+            policy: PolicySet(policies: [
+                OCSPResponderSingingPolicy(issuer: issuer),
+                RFC5280Policy(validationTime: validationTime),
+            ])
+        )
+
+        let validationResult = await verifier.validate(leafCertificate: leafCertificate, intermediates: CertificateStore())
+        let validatedLeaf: Certificate
+        switch validationResult {
+        case .couldNotValidate(let failures):
+            return .failsToMeetPolicy(reason: "could not validate OCSP responder certificates \(failures)")
+        case .validCertificate(let chain):
+            // force unwrap is safe because a valid chain always contains a at least the leaf certificate
+            validatedLeaf = chain.first!
+        }
+        
+        return self.validateSignature(
```

Comment:
Chain building is more expensive than signature validation, so we should probably swap the order of these two operations.